homeassistant/components/volumio/__init__.py

```diff
@@ -43,6 +46,28 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     return True
 
 
+def volumio_exception_handler(func):
+    """Decorate Volumio calls to handle Volumio exceptions."""
+
+    async def handler(self, *args, **kwargs):
+        try:
+            return await func(self, *args, **kwargs)
+        except CannotConnectError as error:
+            if self.available:
+                # Currently this always returns True.
+                # TO DO: Implement better handling of loss of connection.
+                # Combine this with better update handling,
+                # using DataUpdateCoordinator and CoordinatorEntity?
+                # Look at Roku integration for example
```

Comment:
I think tasks/todo's don't like this don't belong in code.
Especially this is probably not the place to implement it either.